pkg/cache/cache.go

```diff
@@ -103,6 +103,10 @@ type Options struct {
 	// Namespace restricts the cache's ListWatch to the desired namespace
 	// Default watches all namespaces
 	Namespace string
+
+	// FieldSelectorByResource restricts the cache's ListWatch to the desired
+	// fields per resource
+	FieldSelectorByResource map[string]string
```

Comment:
What's the key and value expected to look like here? Would it better to use a GroupKind as key instead?

Reply:
Or GroupResource?

Reply:
Used GroupResource so we can match all the versions.